src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceBatchSourceETLTest.java

```diff
@@ -0,0 +1,143 @@
+import co.cask.cdap.datapipeline.SmartWorkflow;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import co.cask.cdap.etl.mock.batch.MockSink;
+import co.cask.cdap.etl.mock.test.HydratorTestBase;
+import co.cask.cdap.etl.proto.v2.ETLBatchConfig;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.etl.proto.v2.ETLStage;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.TestConfiguration;
+import co.cask.cdap.test.WorkflowManager;
+import co.cask.hydrator.salesforce.plugin.source.batch.SalesforceBatchSource;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.internal.AssumptionViolatedException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Methods to run ETL with Salesforce Bulk plugin as source, and a mock plugin as a sink.
+ *
+ * By default all tests will be skipped, since Salesforce credentials are needed.
+ *
+ * Instructions to enable the tests:
+ * 1. Create/use existing Salesforce account
+ * 2. Tables need to have some records. Defaults setup by Salesforce during account creation are enough.
+ * 2. Create connected application within the account to get clientId and clientSecret
+ * 3. Run the tests using the command below:
+ *
+ * mvn clean test -Dsf.test.clientId= -Dsf.test.clientSecret= -Dsf.test.username= -Dsf.test.password=
+ *
+ */
+public abstract class BaseSalesforceBatchSourceETLTest extends HydratorTestBase {
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  private static final ArtifactSummary APP_ARTIFACT = new ArtifactSummary("data-pipeline", "3.2.0");
+
+  // TODO: move to properties
+  // TODO: ignore if not set
+  private static final String CLIENT_ID = System.getProperty("sf.test.clientId");
+  private static final String CLIENT_SECRET = System.getProperty("sf.test.clientSecret");
+  private static final String USERNAME = System.getProperty("sf.test.username");
+  private static final String PASSWORD = System.getProperty("sf.test.password");
+  private static final String LOGIN_URL = System.getProperty("sf.test.loginUrl",
+                                                            "https://login.salesforce.com/services/oauth2/token");
+
+  @BeforeClass
+  public static void setupTestClass() throws Exception {
+    try {
+      Assume.assumeNotNull(CLIENT_ID, CLIENT_SECRET, USERNAME, PASSWORD, LOGIN_URL);
+    } catch (AssumptionViolatedException e) {
+      System.out.println("WARNING: ETL tests are skipped. Please find the instructions on enabling it at" +
```

Comment:
Add message to runtime exception and remove System.out.println

Reply:
We need AssumptionViolatedException so the tests are skipped. Not runtime.

Reply:
The main point here is to remove `System.out.println`.  You can replace it with logger.